gpii/node_modules/flowManager/src/FlowManager.js

```diff
@@ -138,12 +138,37 @@ fluid.defaults("gpii.flowManager.local", {
             type: "gpii.flowManager.getGpiiKey.handler"
         }
     },
+    invokers: {
+        // Needs to be implemented if the handling is required when users update their preferences.
+        setSettings: "fluid.identity"
+    },
+    events: {
+        preferencesSavedSuccess: null,
+        preferencesSavedError: null
+    },
     listeners: {
-        "onCreate": "gpii.singleInstance.registerInstance",
-        "afterDestroy": "gpii.singleInstance.deregisterInstance"
+        "onCreate.registerInstance": "gpii.singleInstance.registerInstance",
+        "afterDestroy.deregisterInstance": "gpii.singleInstance.deregisterInstance",
+        "{lifecycleManager}.events.preferencesUserUpdated": {
+            namespace: "savePreferences",
+            listener: "gpii.flowManager.local.savePreferences",
+            args: ["{that}", "{arguments}.0", "{arguments}.1"]
+        }
     }
 });
 
+gpii.flowManager.local.savePreferences = function (that, gpiiKey, preferences) {
+    var savePromise = that.setSettings(gpiiKey, preferences);
```

Comment:
This "internal polymorphism" is fragile and unnecessary. It's ok to implement setSettings so it always returns a promise - or alternatively just go over to the "transforming promise chain" approach and make setSettings into a pseudoevent as we are now doing in many places.
https://github.com/fluid-project/infusion/blob/master/src/framework/core/js/DataSource.js#L178